include/hc_printf.hpp

```diff
@@ -64,40 +71,33 @@ enum PrintfError {
 
 static inline PrintfPacket* createPrintfBuffer(hc::accelerator& a, const unsigned int numElements) {
   PrintfPacket* printfBuffer = NULL;
-  if (numElements > 6) {
+  if (numElements > 5) {
     printfBuffer = hc::am_alloc(sizeof(PrintfPacket) * numElements, a, amHostCoherent);
 
-    // initialize the printf buffer header
+    // Initialize the Header elements of the Printf Buffer
     printfBuffer[0].type = PRINTF_BUFFER_SIZE;
     printfBuffer[0].data.ui = numElements;
-    printfBuffer[1].type = PRINTF_BUFFER_CURSOR;
-    printfBuffer[1].data.ui = 5;
-    printfBuffer[2].type = PRINTF_STRING_BUFFER;
-    printfBuffer[2].data.ptr = hc::am_alloc(sizeof(char) * numElements * 12, a, amHostCoherent);
-    printfBuffer[3].type = PRINTF_STRING_BUFFER_SIZE;
-    printfBuffer[3].data.ui = numElements * 12;
-    printfBuffer[4].type = PRINTF_STRING_BUFFER_CURSOR;
-    printfBuffer[4].data.ui = 0;
+
+    // Header includes a helper string buffer which holds all char* args
+    printfBuffer[1].type = PRINTF_STRING_BUFFER;
+    printfBuffer[1].data.ptr = hc::am_alloc(sizeof(char) * numElements * 12, a, amHostCoherent);
+    printfBuffer[2].type = PRINTF_STRING_BUFFER_SIZE;
+    printfBuffer[2].data.ui = numElements * 12;
```

Comment:
@aaronenyeshi  i'm a bit afraid of all these magic numbers. could you try change them into enums?